src/rust/src/encoder/g608.rs

```diff
@@ -0,0 +1,308 @@
+use crate::bindings::{
+    ccx_encoding_type_CCX_ENC_ASCII, ccx_encoding_type_CCX_ENC_LATIN_1,
+    ccx_encoding_type_CCX_ENC_UNICODE, ccx_encoding_type_CCX_ENC_UTF_8, eia608_screen, encoder_ctx,
+    font_bits_FONT_ITALICS, font_bits_FONT_REGULAR, font_bits_FONT_UNDERLINED,
+    font_bits_FONT_UNDERLINED_ITALICS,
+};
+use crate::encoder::common::{encode_line, write_raw};
+use crate::libccxr_exports::time::ccxr_millis_to_time;
+use lib_ccxr::util::encoding::{line21_to_latin1, line21_to_ucs2, line21_to_utf8};
+use std::io;
+use std::os::raw::{c_int, c_uchar, c_uint, c_void};
+
+/// Write data to file descriptor with retry logic
+pub fn write_wrapped(fd: c_int, buf: &[u8]) -> Result<(), io::Error> {
+    let mut remaining = buf.len();
+    let mut current_buf = buf.as_ptr();
+
+    while remaining > 0 {
+        let written = write_raw(fd, current_buf as *const c_void, remaining);
+        if written == -1 {
+            return Err(io::Error::last_os_error());
+        }
+        let bytes_written = written as usize;
+        unsafe {
+            current_buf = current_buf.add(bytes_written);
+        }
+        remaining -= bytes_written;
```

Comment:
Why do you need to create another variable here, how about not using it and directly doing
```suggestion
        unsafe {
            current_buf = current_buf.add(bytes_written);
        }
        remaining -= written;
```